Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -472,6 +469,109 @@ public static Vector3 AxisDirection(int axisIndex, Transform givenTransform = nu
             return worldDirections[(int)Mathf.Clamp(axisIndex, 0f, worldDirections.Length)];
         }
 
+        /// <summary>
+        /// The AddListValue method attempts to add a value in the given list if the value does not already exist. `forceAdd = true` will always add a new value to the list even if it already exists.
+        /// </summary>
+        /// <typeparam name="TValue">The datatype for the list value.</typeparam>
+        /// <param name="list">The list to retrieve the value from.</param>
+        /// <param name="value">The value to attempt to add to the list.</param>
+        /// <param name="forceAdd">If this is `true` then the value provided will always be appended to the list. If this is `false` the value provided will only be added to the list if it doesn't already exist.</param>
+        /// <returns>Returns `true` if the given value was successfully added to the list. Returns `false` if the given value already existed in the list and `forceAdd = false`.</returns>
+        public static bool AddListValue<TValue>(ref List<TValue> list, TValue value, bool forceAdd = false)
+        {
+            if (forceAdd || !list.Contains(value))
+            {
+                list.Add(value);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// The RemoveListValue method attempts to remove an entry from the given list based on the given value.
+        /// </summary>
+        /// <typeparam name="TValue">The datatype for the list value.</typeparam>
+        /// <param name="list">The list to remove the value from.</param>
+        /// <param name="value">The value to remove.</param>
+        /// <returns>Returns `true` if the value was found and removed. Returns `false` if the value was not found and therefore not removed.</returns>
+        public static bool RemoveListValue<TValue>(ref List<TValue> list, TValue value)
+        {
+            if (list.Contains(value))
```

Comment:
Remove this and change the whole method to
```c#
return list.Remove(value);
```
instead. Or better: Remove this helper method altogether since it's now adding nothing to the existing method on `List<TValue>`.

Reply:
If the value doesn't exist then won't that throw an exception?

Reply:
Nope, it will instead return true/false like you do:
https://msdn.microsoft.com/en-us/library/cd666k3e(v=vs.90).aspx

Reply:
same as the dictionary then, remove in place of `list.Remove(value)`